File: Interviews/GFG/random-problems/Medium/Sum-of-Beauty-of-All-Substrings/Sum-of-Beauty-of-All-Substrings.cpp
class Solution {
   public:
    int beautySum(string s) {
        int ans = 0;
        for (int i = 0; i < s.size(); i++) {
            vector<int> cnt(26, 0);
            for (int j = i; j < s.size(); j++) {
                cnt[s[j] - 'a'] += 1;
                int maxx = 0, minn = 1000;
                for (int k = 0; k < 26; k++) {
                    maxx = max(maxx, cnt[k]);
                    if (cnt[k] > 0) {
                        minn = min(minn, cnt[k]);
                    }
                }

                ans += maxx - minn;
            }
        }

        return ans;
    }
};

File: Interviews/GFG/Daily-Problems/2022/November/Day-16/Sum-of-Beauty-of-All-Substrings.py
class Solution:
    def beautySum(self, s):
        ans = 0
        for idx, i in enumerate(s):
            cnt = [0 for i in range(26)]
            for j in range(idx, len(s)):
                cnt[ord(s[j]) - 97] += 1
                maxx, minn = 0, 1000
                for k in range(26):
                    maxx = max(maxx, cnt[k])
                    if cnt[k] > 0:
                        minn = min(minn, cnt[k])
                ans += maxx - minn
        return ans
